### PR TITLE
Rum 18631 mandatory fields

### PR DESCRIPTION
**Pull request by MalcolmSmithUK**

### CODE REVIEW

File: tests/rci.spec.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
+const localStorageClientIdKey = "eggplantRciClientId";
```
Comment (robertdumitrescu) [inline tests/rci.spec.js]: Port the tests from mocha/chai to jest

File: src/collectors/utility/uuid.js [hunk 2]
```diff
@@ -1,25 +1,28 @@
...
-	    uuid += '-';
-	    uuid += (Math.random() * 4 | 8).toString(16);
-	    break;
-	  default:
-	    uuid += (Math.random() * 16 | 0).toString(16);
-	  }
-	}
+  // return uuid of form xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx
+  let uuid = ''; let
+    ii;
```
Comment (yghern) [inline src/collectors/utility/uuid.js]: [FORMAT] strange line break

File: src/core/Collector.js [hunk 3]
```diff
@@ -1,32 +1,21 @@
-export class Collector {
-	constructor(transport, collectors) {
-		this.transport = transport;
-		this.collectors = Array.isArray(collectors) ? [].concat(collectors) : [];
-	}
+/* eslint no-await-in-loop: 0 */
```
Comment (yghern) [inline src/core/Collector.js]: why this file is exception from eslint rule? is there no other way?

File: tests/rci.spec.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        expect(window.assertOnMe.clientId).equal(localStorage.getItem(localStorageClientIdKey));
+        expect(window.assertOnMe.deviceType).to.not.equal(null);
+        expect(window.assertOnMe.deviceType).to.be.a('string');
+        expect(window.assertOnMe.errorCode).equal(expectedErrorCode);
+        expect(window.assertOnMe.errorType).equal(expectedErrorType);
+        expect(window.assertOnMe.errorFatal).equal(expectedErrorFatal);
+        expect(window.assertOnMe.eventAction).equal(expectedEventAction);
+        expect(window.assertOnMe.eventType).equal(expectedEventType);
+        expect(window.assertOnMe.id).to.not.equal(null);
+        expect(window.assertOnMe.journeyAction).equal('stop');
```
Comment (yghern) [inline tests/rci.spec.js]: We have global now for journey action (JOURNEY_ACTION)

**Review by robertdumitrescu — APPROVED** [review]

**Review by yghern — CHANGES_REQUESTED** [review]

**Review by yghern — APPROVED** [review]

**Pull request merged**